internal/controller/grafanaorganization_controller.go

```diff
@@ -100,20 +92,82 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 		}
 	}
 
-	// Ensure the first organization is renamed.
-	_, err := grafanaAPI.Orgs.UpdateOrg(1, &grafanaAPIModels.UpdateOrgForm{
-		Name: "Shared Org.",
-	})
-	if err != nil {
-		logger.Error(err, "Could not rename Main Org. to Shared Org.")
-		return ctrl.Result{}, errors.WithStack(err)
+	// if the CR doesn't have an orgID, create the organization in Grafana and update the status
+	if grafanaOrganization.Status.OrgID == 0 {
+		return ctrl.Result{}, r.createOrganizationInGrafana(ctx, grafanaAPI, grafanaOrganization)
+	} else {
+		grievious, err := grafanaAPI.Orgs.GetOrgByID(grafanaOrganization.Status.OrgID)
```

Comment:
I Hope you'll rename those grievous and Obi-Wan variables ?

Reply:
Definitely :sweat_smile: